app/bundles/PageBundle/EventListener/SegmentTrackingSubscriber.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\PageBundle\EventListener;
+
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Entity\LeadListRepository;
+use Mautic\PageBundle\Event\UrlTokenReplaceEvent;
+use Mautic\PageBundle\PageEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Appends contact segment IDs to tracking URLs for third-party integrations like VWO.
+ */
+class SegmentTrackingSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private CoreParametersHelper $coreParametersHelper,
+        private LeadListRepository $leadListRepository,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            PageEvents::ON_URL_TOKEN_REPLACE => ['onUrlTokenReplace', -100],
```

Comment:
If we ditch the artificial event name then this will be changed like so:
```suggestion
            UrlTokenReplaceEvent::class => ['onUrlTokenReplace', -100],
```